File: day1/problem1.py
sample1 = open('./inputs/sample1.txt', 'r').read()
input1 = open('./inputs/input1.txt', 'r').read()


def solution(input_txt):
    totals = []
    lines = input_txt.split('\n')
    cur_sum = 0

    for lineStr in lines:
        if lineStr != '':
            cur_sum += int(lineStr)
        else:
            totals.append(cur_sum)
            cur_sum = 0

    if cur_sum > 0:
        totals.append(cur_sum)

    return max(totals)


print('sample:', solution(sample1))
print('solution:', solution(input1))

File: day1/problem1.js
const sample1 = require('fs').readFileSync(require('path').resolve(__dirname, './inputs/sample1.txt'), 'utf8');
const input1 = require('fs').readFileSync(require('path').resolve(__dirname, './inputs/input1.txt'), 'utf8');

const solution = (input) => {
  let maxCalories = 0;

  input
    .split('\n')
    .map(Number)
    .reduce((elfSum, cal) => {
      maxCalories = Math.max(maxCalories, elfSum + cal);
      return cal ? elfSum + cal : 0;
    }, 0);

  return maxCalories;
};

console.log('sample:', solution(sample1));
console.log('solution:', solution(input1));
